### PR TITLE
Duplicate child suites get added when working with parent/child suite scenario

### LINKED ISSUE #1533: Duplicate child suites get added when working with parent/child suite scenario
**TestNG 6.12**

### Expected behavior
Duplicate child suites should not get added.

### Actual behavior
Duplicate child suites is getting added.

### Is the issue reproductible on runner?

- [X] Shell
- [X] Maven
- [X] Gradle
- [X] Ant
- [X] Eclipse
- [X] IntelliJ
- [X] NetBeans

**Sample Test class** 

```java
import org.testng.annotations.Test;

public class SampleTestClass {
    @Test
    public void testMethod() {
        System.err.println("Hello World");
    }
}
```

**Child Suite xml**

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="46125558_Suite" verbose="2">
    <test name="46125558_Test">
        <classes>
            <class name="com.rationaleemotions.stackoverflow.qn46125558.SampleTestClass"/>
        </classes>
    </test>
</suite>
```

**Parent Suite xml**

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="46125558_Parent_Suite" verbose="2">
    <suite-files>
        <suite-file path="child.xml"/>
    </suite-files>
</suite>
```

**Testcase that can be executed to reproduce the problem**

```java
import org.testng.Assert;
import org.testng.ISuite;
import org.testng.ISuiteListener;
import org.testng.TestNG;
import org.testng.xml.Parser;
import org.testng.xml.XmlSuite;
import org.xml.sax.SAXException;

import javax.xml.parsers.ParserConfigurationException;
import java.io.IOException;
import java.util.Collections;
import java.util.List;
import java.util.concurrent.atomic.AtomicInteger;

public class LocalXmlSuiteParser {
    public static void main(String[] args) throws IOException, SAXException, ParserConfigurationException {
        String suiteFile = "src/test/resources/46125558/parent.xml";
        List<XmlSuite> suites = new Parser(suiteFile).parseToList();
        Assert.assertEquals(suites.size(), 1);
        Assert.assertEquals(suites.get(0).getChildSuites().size(), 1);
        TestNG testng = new TestNG();
        testng.setXmlSuites(suites);
        testng.setListenerClasses(Collections.singletonList(SuiteCounter.class));
        testng.run();
        Assert.assertEquals(SuiteCounter.counter.get(), 2);
    }

    public static class SuiteCounter implements ISuiteListener {
        static AtomicInteger counter = new AtomicInteger(0);

        @Override
        public void onStart(ISuite suite) {
            counter.incrementAndGet();
        }

        @Override
        public void onFinish(ISuite suite) {

        }
    }
}
```

For additional context, please refer to [this](https://stackoverflow.com/questions/46125558/testng-running-multiple-suites-running-multiple-times) StackOverFlow thread.

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #1533

Fixes #1533 .

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

**Comment by krmahadevan** [comment]
ping @juherr - Please help take a look

**Comment by selenareneephillips** [comment]
Looks like there are some failures in the parallelization tests. I can't tell what is going on though. The information is in the log statements which aren't in the build report. :-/. Has there been any headway on getting some real visibility into the test results or is it still only possible to get that information with system.out.println statements?

**Comment by krmahadevan** [comment]
@selenareneephillips - Yes, most of the times the failures in the CI runs are related to parallelized tests. I haven't been able to make much of a progress in terms of making these test results available outside of travis. I will try and figure out something perhaps in a day or two.

**Comment by juherr** [comment]
#1526 may be a solution

**Comment by juherr** [comment]
@selenareneephillips @krmahadevan Fyi, you should be able to rerun build on Travis once authenticated with your GitHub credentials.

File: src/test/java/test/suites/github1533/Github1533Test.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package test.suites.github1533;
```
Comment (juherr) [inline src/test/java/test/suites/github1533/Github1533Test.java]: Maybe `test.xml.github1533`

**Review by juherr — APPROVED** [review]
The change is fixing the issue but not the design: `XmlSuite` should be only the java representation of the xml file and should don't mix values and state. Another problem is the new method will be in the public api: just add a comment "for internal use" or a similar thing.

**Review by cbeust — APPROVED** [review]

**Comment by krmahadevan** [comment]
@cbeust - You have approved the PR but yet to be merged. Please let me know if there is anything outstanding from my side.

**Comment by cbeust** [comment]
I was actually not sure if you wanted to make additional changes first so I approved it so you could merge it yourself when it's ready.

**Comment by krmahadevan** [comment]
@cbeust Ah ok. I was confused because first time I saw you approved a PR but not merged it. Please let me know if it's fine for us to merge the code once its been peer reviewed and additionally approved by you.

**Comment by cbeust** [comment]
Absolutely, you're welcome to merge once I've approved it.

**Pull request merged**